agreement/abstractions.go

```diff
@@ -229,6 +229,10 @@ type KeyManager interface {
 	// valid for the provided votingRound, and were available at
 	// keysRound.
 	VotingKeys(votingRound, keysRound basics.Round) []account.Participation
+
+	// Record indicates that the given participation action has been taken.
+	// This function is also part of account.ParticipationRegistry.
+	Record(account basics.Address, round basics.Round, participationType account.ParticipationAction) error
```

Comment:
I'm wondering if we really need to have this `error` return variable. I can't think of anything useful we can do with it. Can you ? i.e. having the the `error` between the `participationRegistry` and the `AlgorandFullNode` makes sense, since we want the node to log all the errors. However, providing that feedback to the agreement isn't very helpful since we don't have any useful thing we can do with that.
( i.e. just ignoring it implies that the interface is wrong.. )

Reply:
I hadn't thought about it like that. I think that the `ParticipationRegistry` interface can return the error, and the agreement abstraction can remove it. In the shim function implemented by the FullNode I'll log a warning and the agreement code can ignore that case.